src/components/RangeDatePicker/BaseRangeDatePicker.tsx

```diff
@@ -0,0 +1,285 @@
+import React from 'react'
+import { CSSProperties, Ref, useEffect, useRef, useState } from 'react'
+
+import { ExternalStyles, focusBoxShadow, Theme, useStyles } from '../../styles'
+import { composeRefs } from '../../util/react'
+import { DateInput } from '../DateField'
+import { Icons } from '../Icon'
+import { InputWrapper } from '../TextField/InputWrapper'
+
+export interface Period {
+  startDate?: Date
+  finalDate?: Date
+}
+
+export interface BaseRangeDatePickerProps {
+  /**
+   * Prop used to set an initial period.
+   */
+  value?: Period
+
+  /**
+   *  Props to control limited periods
+   */
+  minDate?: Date
+  maxDate?: Date
+
+  /**
+   * Prop to disable the date field.
+   */
+  disabled?: boolean
+
+  /**
+   * Prop to set an icon in date field button.
+   * When this prop is not declared, date field will be
+   * shown without a icon.
+   */
+  icon?: Icons
+
+  /**
+   *  Prop to reference an invalid state.
+   */
+  invalid?: boolean
+
+  /**
+   *  Prop for receive external styles.
+   */
+  style?: ExternalStyles
+
+  /**
+   *  Props to set placeholder to inputs
+   */
+  startPlaceholder?: string
+  finalPlaceholder?: string
+
+  /**
+   * initialInputRef and finalInputRef are used to assign refs
+   * to date input components
+   */
+  initialInputRef?: Ref<HTMLInputElement>
+  finalInputRef?: Ref<HTMLInputElement>
+
+  /**
+   * Function to manage the icon action
+   */
+  onIconClick?(): void
+
+  /**
+   * Function to control what input is on focus
+   *
+   * @param isOnFocus
+   */
+  onInputOnFocus?(isOnFocus: number): void
+
+  /**
+   * Function used to manipulate values of Period
+   *
+   * @param period
+   */
+  onChange?(period: Period): void
+}
+
+export function BaseRangeDatePicker(props: BaseRangeDatePickerProps) {
```

Comment:
In my view, the name of this component should not include the "picker" term. The "picker" is the popup calendar that is opened and this component is just the two base inputs. Maybe `RangeDateInput`?